src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -0,0 +1,369 @@
+                                "information_schema.tables.table_name = information_schema.columns.table_name " +
+                                "WHERE information_schema.tables.table_schema='$PUBLIC_SCHEMA' " +
+                                "AND table_type='BASE TABLE' " +
+                                "ORDER BY information_schema.tables.table_name")
+                while (rs.next()) {
+                    val tableName = rs.getString("table_name")
+                    val columnName = rs.getString("column_name")
+                    columnNamesByTableName.getOrPut(tableName) { mutableSetOf() }.add(columnName)
+                }
+            }
+
+        }
+        return columnNamesByTableName
+    }
+
+    fun createNewColumnObjects(dbName: String, tableName: String, tableId: UUID, orgId: UUID, columnName: Optional<String>): Set<OrganizationExternalDatabaseColumn> {
+        val columnCondition = if (columnName.isPresent) {
+            "AND information_schema.columns.column_name = '${columnName.get()}'"
+        } else {
+            ""
+        }
+        val newColumns = mutableSetOf<OrganizationExternalDatabaseColumn>()
+        assemblerConnectionManager.connect(dbName).use { dataSource ->
+            dataSource.connection.createStatement().use { stmt ->
+                val rs = stmt.executeQuery(
+                        "SELECT information_schema.tables.table_name, information_schema.columns.column_name, " +
+                                "information_schema.columns.data_type, information_schema.columns.ordinal_position, " +
+                                "information_schema.table_constraints.constraint_type " +
+                                "FROM information_schema.tables " +
+                                "LEFT JOIN information_schema.columns on information_schema.tables.table_name = " +
+                                "information_schema.columns.table_name " +
+                                "LEFT OUTER JOIN information_schema.constraint_column_usage on " +
+                                "information_schema.columns.column_name = information_schema.constraint_column_usage.column_name " +
+                                "LEFT OUTER JOIN information_schema.table_constraints " +
+                                "on information_schema.constraint_column_usage.constraint_name = " +
+                                "information_schema.table_constraints.constraint_name " +
+                                "WHERE information_schema.columns.table_name = '$tableName' " +
+                                "$columnCondition"
+                )
+                while (rs.next()) {
+                    val columnName = rs.getString("column_name")
+                    val dataType = rs.getString("data_type")
+                    val position = rs.getInt("ordinal_position")
+                    val isPrimaryKey = rs.getString("constraint_type") == "PRIMARY KEY"
```

Comment:
This should be in a dedicated field (`"PRIMARY KEY"`) or in an enum `ConstraintType`

Reply:
cool i'll change this